test/boost/memtable_test.cc

```diff
@@ -814,3 +815,90 @@ SEASTAR_TEST_CASE(failed_flush_prevents_writes) {
+                service::migration_manager& mm = env.migration_manager().local();
+                auto group0_guard = co_await mm.start_group0_operation();
+                auto ts = group0_guard.write_timestamp();
+                auto announcement = co_await mm.prepare_new_column_family_announcement(s, ts);
+                co_await mm.announce(std::move(announcement), std::move(group0_guard));
+            }
+
+            future<result_t> operator()() {
+                const sstring ks_name = this->ks_name();
+                const sstring cf_name = this->cf_name(this_shard_id());
+                random_mutation_generator gen{
+                    random_mutation_generator::generate_counters::no,
+                    local_shard_only::yes,
+                    random_mutation_generator::generate_uncompactable::no,
+                    std::nullopt,
+                    ks_name.c_str(),
+                    cf_name.c_str()
+                };
+                schema_ptr s = gen.schema();
+
+                co_await create_table(s);
+                replica::database& db = env.local_db();
+                replica::table& t = db.find_column_family(ks_name, cf_name);
+
+                size_t max_sstables_count = 0;
+                semaphore sem{num_concurrent_flushes};
+                unsigned mutations_applied = 0;
+                auto range = boost::irange<int>(0, num_flushes);
+                co_await seastar::parallel_for_each(range.begin(), range.end(), [&](int value) -> future<> {
+                    auto permit = co_await get_units(sem, 1);
+                    std::vector<mutation> mutations = gen(num_mutations_per_flush);
+                    for (auto&& m : mutations) {
+                        auto sync = mutations_applied % mutations_per_commitlog_sync == 0 ? db::commitlog::force_sync::yes : db::commitlog::force_sync::no;
+                        co_await db.apply(t.schema(), freeze(m), tracing::trace_state_ptr(), sync, db::no_timeout);
+                    }
+                    co_await t.flush();
+                    max_sstables_count = std::max(t.sstables_count(), max_sstables_count);
+                });
+                co_return std::make_tuple(this_shard_id(), max_sstables_count, t.schema()->max_compaction_threshold() * 2);
+            }
+        };
+
+        replica::database& db = env.local_db();
+        const unsigned thread_count = smp::count;
+
+        std::vector<unsigned> concurrent_flushes{2, 2};
+        std::vector<unsigned> mutations_per_flush{1, 400};
+        std::vector<unsigned> num_flushes{500, 10};
+        std::vector<future<result_t>> futures;
+        for (unsigned thread_id : boost::irange<unsigned>(0, thread_count)) {
+            auto idx = thread_id % 2;
+            futures.push_back(smp::submit_to(thread_id, flusher{.env=env, .num_flushes=num_flushes[idx], .num_mutations_per_flush=mutations_per_flush[idx], .num_concurrent_flushes=concurrent_flushes[idx]}));
```

Comment:
Why are you involving submit_to() in the test? Doesn't it make it dependent on the number of shards?

Reply:
We need at least two threads, so that one can temporarily block the other to simulate the slow compaction scenario (the comment below https://github.com/scylladb/scylla/pull/10717#discussion_r894270243 describes the details). I'll add a check to assert the number of threads.